src/TestFramework/TestFramework/Assertions/Assert.IsInstanceOfType.cs

```diff
@@ -131,6 +149,18 @@ public static void IsInstanceOfType<T>([NotNull] object? value, string? message)
     public static void IsInstanceOfType<T>([NotNull] object? value, string? message, params object?[]? parameters)
         => IsInstanceOfType(value, typeof(T), message, parameters);
 
+    /// <summary>
+    /// Tests whether the specified object is an instance of the generic
+    /// type and throws an exception if the generic type is not in the
+    /// inheritance hierarchy of the object.
+    /// </summary>
+    /// <typeparam name="T">The expected type of <paramref name="value"/>.</typeparam>
+    public static void IsInstanceOfType<T>([NotNull] object? value, out T instance, string? message, params object?[]? parameters)
+    {
+        IsInstanceOfType(value, typeof(T), message, parameters);
+        instance = (T)value!;
```

Comment:
Value is validated as not-null so you can do

```suggestion
        instance = (T)value;
```